immuni_common/helpers/utils.py

```diff
@@ -52,3 +53,32 @@ def dense_dict(dictionary: Dict) -> Dict:
         for k, v in dictionary.items()
         if v is not None
     }
+
+
+# Note: This could be a Generic[T] with T as TypeVar. That would allow
+#  us to define a WeightedPair[str] and have the weighted_random
+#  return a valid typed str when used, but generic NamedTuples are not yet
+#  supported by mypy:
+#  https://github.com/python/mypy/issues/685
+class WeightedPair(NamedTuple):
+    """
+    Simple NamedTuple that allows us to describe
+    weighted key/value pairs for weighted random.
+    """
+
+    weight: int
+    payload: Any
```

Comment:
From here I can deduce it's tailored to responses, thus I'd go for `WeightedResponse`.

Reply:
Why? Payload could be anything. I would make it generic if not for the bug in mypy 😞

Reply:
Yet, there's no point in making it generic right in this project? 🤔 We don't have foreseeable features that would require it, AFAIK.

Reply:
No, but I don't see why we should make it tailored for responses 🤔. Honestly I think the argument for making it NOT tailored is just slightly better, since it it allows more flexibility, should we ever need it.